src/dbnode/storage/bootstrap.go

```diff
@@ -228,16 +233,54 @@ func (m *bootstrapManager) bootstrap() error {
 		}
 	}()
 
-	targets := make([]bootstrap.ProcessNamespace, 0, len(namespaces))
+	start := m.nowFn()
+	m.log.Info("bootstrap prepare")
+
+	var (
+		bootstrapNamespaces = make([]bootstrapNamespace, len(namespaces))
+		prepareWg           sync.WaitGroup
+		prepareLock         sync.Mutex
+		prepareMultiErr     xerrors.MultiError
+	)
+	for i, namespace := range namespaces {
+		i, namespace := i, namespace
+		prepareWg.Add(1)
+		go func() {
+			defer prepareWg.Done()
+
+			shards, err := namespace.PrepareBootstrap()
+
+			prepareLock.Lock()
+			defer prepareLock.Unlock()
+
+			if err != nil {
+				prepareMultiErr = prepareMultiErr.Add(err)
```

Comment:
nit: worth logging the error here?

Reply:
I do that elsewhere with I believe with `"bootstrap prepare failed"` - to avoid log spam was thinking just doing this once.